rollup.config.js

```diff
@@ -122,15 +125,85 @@ function resolveGenerated() {
     name: "resolve-msg-generated",
     resolveId(importee) {
       if (importee.startsWith("gen/msg_generated")) {
-        const resolved = path.resolve(
-          path.join(process.cwd(), `${importee}.ts`)
-        );
-        return resolved;
+        return path.resolve(`${importee}.ts`);
       }
     }
   };
 }
 
+function generateDepFile({ outputFile, sourceFiles = [], configFiles = [] }) {
+  let timestamp = Date.now();
+
+  // Save the depfile just before the node process exits.
+  process.once("beforeExit", () =>
+    writeDepFile({ outputFile, sourceFiles, configFiles, timestamp })
+  );
```

Comment:
Is this the only hook to do this sort of thing? What if there's an error in rollup?

Reply:
🤷‍♂️  ... node will crash before the depfile gets written I suppose?